### DISCUSSION TITLE
Firebase Auth with Vercel deployment redirects to 404 page

### DISCUSSION THREAD

**Discussion by MartinXPN** [gladly-team/next-firebase-auth#537] [Q&A] [answered]
I was using React to deploy my app through firebase hosting. I've switched to Next.js and Vercel deployment now.
The problem I'm encountering is that Next.js redirects the auth handler URL to a 404 page instead of the authentication handler popup.

The app is available at: https://profound.academy
The authentication handler is: https://profound.academy/__/auth/handler

I'm using Google, Facebook, Microsoft, and email/password authentication. The view is built with Firebase UI Web.

My pages directory is as follows:
![Screen Shot 2022-08-23 at 11 30 55 AM](https://user-images.githubusercontent.com/16335716/186098285-bccabc57-9b5b-47d5-b475-77b15a6333e3.png)

Inside `[courseId]/index.ts`, I return `{notFound: true}` if there are no courses available in the database with that id. Yet, I would expect the `/__/auth/handler` route to be automatically redirected to the authentication popup page.

Is there any way to whitelist this route so that the authentication works? It currently shows a 404 page which makes it impossible for users to log into the platform.

I've currently changed the `authDomain` as suggested here: https://stackoverflow.com/a/70072471/5512770
But it would be way better if the app would work properly on Vercel and would show `profound.academy` on the login page instead of `profound-academy.web.app`.

**Comment by kmjennison** [comment, marked as answer]
Unless I'm misunderstanding, this doesn't appear to be related to `next-firebase-auth`. This sounds like a question better suited for Next.js, Vercel, Firebase, or Firebase UI Web discussions.

**Reply by MartinXPN** [reply]
Yeah, I guess you're right. Thanks for the response. I'll move this to their repo.
